tests/integration/test_records.py

```diff
@@ -56,43 +57,54 @@ def test_float_vector(client):
     ensemble_id = _create_ensemble(client)
 
     records = [
-        ("name", "[1, 2, 3, 4, 5]", True),
-        ("name", "[5, 4, 3, 2, 1]", False),
-        ("fail", "yes", False),
+        ("name", "[1, 2, 3, 4, 5]", status.HTTP_200_OK),
+        ("name", "[5, 4, 3, 2, 1]", status.HTTP_409_CONFLICT),
+        ("fail", "yes", status.HTTP_415_UNSUPPORTED_MEDIA_TYPE),
     ]
 
     # Post each of the record
-    for name, data, should_succeed in records:
+    for name, data, status_code in records:
         failed = False
         try:
             resp = client.post(
                 f"/ensembles/{ensemble_id}/records/{name}",
                 data=data,
                 headers={"RecordType": "float_vector"},
             )
-            failed = resp.status_code != 200
-        except Exception as exc:
-            failed = True
-        if should_succeed and failed:
-            raise exc
-        elif not should_succeed and not failed:
-            raise AssertionError(
-                f"Posting '{data}' to record '{name}' was expected to fail, but it succeeded"
-            )
+        except Exception:
+            if status_code == status.HTTP_200_OK:
+                raise AssertionError(
+                    f"Posting '{data}' to record '{name}' was expected to succeed, but it failed"
+                )
+
+        assert resp.status_code == status_code
+        if status_code == status.HTTP_200_OK:
+            resp = client.get(f"/ensembles/{ensemble_id}/records/{name}")
+            assert resp.json() == json.loads(data)
+        else:
+            assert "detail" in resp.json()
+            assert resp.json()["detail"]["ensemble_id"] == ensemble_id
+            assert resp.json()["detail"]["name"] == name
+            assert resp.json()["detail"]["error"] != ""
 
     # Compare list of records
     resp = client.get(f"/ensembles/{ensemble_id}/records")
     assert resp.json() == [
         rec[0] for rec in records if rec[2]  # name  # should_succeed
```

Comment:
Test is failing here, because you need to test `rec[2] == status.HTTP_200_OK` rather than just `rec[2]`.

Reply:
That's weird as I've updated it already :thinking: